### PR TITLE
feat(Pagination): new Pagination component

### PR DESCRIPTION
**Pull request by zzlune**
## Checklist

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

### CODE REVIEW

**Comment by AngeliaQ** [comment]
Task linked: [CU-3tvnwm Pagination component](https://app.clickup.com/t/3tvnwm)

File: components/pagination/pagination-item.tsx [hunk 1]
```diff
@@ -19,10 +20,10 @@ const PaginationItem: React.FC<React.PropsWithChildren<PaginationItemProps>> = (
...
 }) => {
   const theme = useTheme()
-  const [hover, activeHover] = useMemo(
-    () => [addColorAlpha(theme.palette.success, 0.1), addColorAlpha(theme.palette.success, 0.8)],
-    [theme.palette.success],
-  )
+  const { variant } = usePaginationContext()
+  const paginationColors = useMemo(() => {
+    return getColors(theme.palette, variant, active)
+  }, [theme.palette, variant, disabled, active])
```
Comment (0x74616e67) [inline components/pagination/pagination-item.tsx]: "disabled" may remove from dependance list

File: components/pagination/pagination-pageSize.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    const pageSize = Number(val)
+    const newPageCount = getPageCount(total, pageSize)
+    const newCurrent = current > newPageCount ? newPageCount : current
+    setPage(newCurrent)
+    setPageSize(pageSize)
+    onPageSizeChange && onPageSizeChange(newCurrent, pageSize)
+  }
+  return (
+    <div className="pagination-pagesize">
+      <span className="text left">SHOW</span>
```
Comment (0x74616e67) [inline components/pagination/pagination-pageSize.tsx]: "SHOW" string can not config.

File: components/pagination/pagination-pageSize.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        initialValue={placeHolderVal}>
+        {pageSizeOptions?.map(pageSize => {
+          return (
+            <Select.Option value={pageSize} key={pageSize}>
+              {pageSize}{' '}
+            </Select.Option>
+          )
+        })}
+      </Select>
+      <span className="text right">RECORDS</span>
```
Comment (0x74616e67) [inline components/pagination/pagination-pageSize.tsx]: "RECORDS" string can not config.

File: components/pagination/pagination-pageSize.tsx (hunk 2 above)
Comment (zzlune) [reply components/pagination/pagination-pageSize.tsx]: Do you mean the 'show' string should be internationalized?

File: components/pagination/pagination-pageSize.tsx (hunk 3 above)
Comment (zzlune) [reply components/pagination/pagination-pageSize.tsx]: Do you mean the 'RECORDS' string should be internationalized?

Comment (0x74616e67) [reply components/pagination/pagination-pageSize.tsx]: Yes, I think it's better to config string like "SHOW" and "RECORDS" through props, user can customize, like:
```
<Pagination itemStringConfig={{
  show: <span>SHOW</span>,
  records: '条数'
}}>
```

File: components/pagination/pagination-item.tsx [hunk 4]
```diff
@@ -38,11 +39,11 @@ const PaginationItem: React.FC<React.PropsWithChildren<PaginationItemProps>> = (
       </button>
       <style jsx>{`
         li {
-          margin-right: 6px;
+          margin-right: 0.5rem;
```
Comment (yqrashawn) [inline components/pagination/pagination-item.tsx]: Better use `layout.gapHalf` and `calc` here.

File: components/pagination/pagination-pageSize.tsx (hunk 3 above)
Comment (zzlune) [reply components/pagination/pagination-pageSize.tsx]: Ok~we will support internationalization instead of customizing elements

**Review by yqrashawn — CHANGES_REQUESTED** [review]

File: components/pagination/pagination-item.tsx (hunk 4 above)
Comment (zzlune) [reply components/pagination/pagination-item.tsx]: done

File: components/pagination/pagination-item.tsx (hunk 1 above)
Comment (zzlune) [reply components/pagination/pagination-item.tsx]: done

**Review by yqrashawn — DISMISSED** [review]

**Review by 0x74616e67 — DISMISSED** [review]

**Review by AngeliaQ — DISMISSED** [review]

File: components/pagination/pagination.tsx [hunk 5]
```diff
@@ -29,6 +29,10 @@ interface Props {
   size?: NormalSizes
   defaultPageSize?: number
   pageSizeOptions?: string[]
+  labelPageSizeChangerBefore?: ReactNode | string
```
Comment (AngeliaQ) [inline components/pagination/pagination.tsx]: is the name too long？ how about labelPageSizeBefore labelJumptoBefore

**Review by AngeliaQ — CHANGES_REQUESTED** [review]

File: pages/en-us/components/pagination.mdx [hunk 6]
```diff
@@ -156,22 +156,28 @@ A long list can be divided into several pages, and only one page will be loaded
...
+| **size**                       | Pagination size                                                                                           | `NormalSizes`                            | [NormalSizes](#normalsizes)               | `medium`                  |
+| **defaultPageSize**            | Default number of data items per page                                                                     | `number`                                 | -                                         | 10                        |
+| **pageSizeOptions**            | Specify the sizeChanger options                                                                           | `string[]`                               | -                                         | ['10', '20', '50', '100'] |
+| **labelPageSizeChangerBefore** | Customize the label before pageSizeChanger                                                                | `ReactNode` / `string`                   | -                                         | `SHOW`                    |
```
Comment (AngeliaQ) [inline pages/en-us/components/pagination.mdx]: need format

**Review by AngeliaQ — CHANGES_REQUESTED** [review]

File: pages/zh-cn/components/pagination.mdx [hunk 7]
```diff
@@ -166,6 +166,10 @@ export const meta = {
 | **size**                | 组件的大小                 | `NormalSizes`                            | [NormalSizes](#normalsizes)               | `medium`                  |
 | **defaultPageSize**     | 默认的每页条数              | `number`                                 | -                                         | 10                        |
 | **pageSizeOptions**     | 指定每页可以显示多少条       | `string[]`                               | -                                         | ['10', '20', '50', '100'] |
+| **labelPageSizeChangerBefore** | 自定义pageSizeChanger组件之前的文本  | `ReactNode` / `string`    | -                                         | `SHOW`                    |
```
Comment (AngeliaQ) [inline pages/zh-cn/components/pagination.mdx]: the same as above

**Review by AngeliaQ — CHANGES_REQUESTED** [review]

File: pages/en-us/components/pagination.mdx (hunk 6 above)
Comment (zzlune) [reply pages/en-us/components/pagination.mdx]: done

File: components/pagination/pagination.tsx (hunk 5 above)
Comment (zzlune) [reply components/pagination/pagination.tsx]: done

File: pages/zh-cn/components/pagination.mdx (hunk 7 above)
Comment (zzlune) [reply pages/zh-cn/components/pagination.mdx]: done

File: components/pagination/pagination-context.ts [hunk 8]
```diff
@@ -1,16 +1,29 @@
...
+  updatePage?: (type: PaginationUpdateType, val?: number) => void
+  updatePageSize?: Function
+  variant?: PaginationVariants
+  page?: number
+  pageSize?: number
 }
 
-const defaultContext = {}
+export interface PaginationHandles {
+  setPage: (value?: number) => void
```
Comment (0x74616e67) [inline components/pagination/pagination-context.ts]: Parameter "value" may be not optional?

File: components/pagination/pagination-context.ts [hunk 9]
```diff
@@ -1,16 +1,29 @@
...
+  getPage: () => number
+  setPageSize: (value?: number) => void
```
Comment (0x74616e67) [inline components/pagination/pagination-context.ts]: Parameter "value" may be not optional?

File: .github/CODEOWNERS [hunk 10]
```diff
@@ -27,4 +28,4 @@ pages/*/components/button.mdx @chain-txz
 pages/*/components/tooltip.mdx @yqrashawn
 pages/*/components/popover.mdx @yqrashawn
 pages/*/components/tag.mdx @yqrashawn
-pages/*/components/tabs.mdx @GuichiZhao
\ No newline at end of file
+pages/*/components/tabs.mdx @GuichiZhao
```
Comment (AngeliaQ) [inline .github/CODEOWNERS:33]: miss mdx codeowner

File: components/pagination/pagination-context.ts (hunk 8 above)
Comment (zzlune) [reply components/pagination/pagination-context.ts]: done

File: components/pagination/pagination-context.ts (hunk 9 above)
Comment (zzlune) [reply components/pagination/pagination-context.ts]: resolved

**Review by 0x74616e67 — DISMISSED** [review]

**Review by AngeliaQ — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 10 above)
Comment (zzlune) [reply .github/CODEOWNERS:33]: resolved

**Review by 0x74616e67 — APPROVED** [review]

**Review by AngeliaQ — APPROVED** [review]

**Pull request merged**